datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -251,6 +252,147 @@ pub async fn from_substrait_plan(
     }
 }
 
+/// An ExprContainer is a container for a collection of expressions with a common input schema
+///
+/// In addition, each expression is associated with a field, which defines the
+/// expression's output.  The data type and nullability of the field are calculated from the
+/// expression and the input schema.  However the names of the field (and its nested fields) are
+/// derived from the Substrait message.
+pub struct ExprContainer {
+    /// The input schema for the expressions
+    pub input_schema: DFSchemaRef,
+    /// The expressions
+    ///
+    /// Each item contains an expression and the field that defines the expected nullability and name of the expr's output
+    pub exprs: Vec<(Expr, Field)>,
+}
+
+// Substrait fields (DataType, nullable) typically don't have names.  In the few cases they do
+// those names are provided as Vec<String>.  This method attaches names to a (DataType, nullable)
+// pair to create a Field
+fn field_from_type_and_names<S: Into<String>, T: Iterator<Item = S>>(
```

Comment:
This feels similar to `from_substrait_type`, could we reuse that somehow?

Reply:
Or actually maybe `make_renamed_schema` is what you want?

Reply:
Good catch.  I found the existing method in the producer but missed it in the consumer.  I've updated the PR to use `make_renamed_schema`'s helper method (which I've renamed as `rename_field`)